src/libsystemd/sd-bus/bus-objects.c

```diff
@@ -1610,6 +1610,39 @@ static int vtable_member_compare_func(const struct vtable_member *x, const struc
 
 DEFINE_PRIVATE_HASH_OPS(vtable_member_hash_ops, struct vtable_member, vtable_member_hash_func, vtable_member_compare_func);
 
+static bool names_are_valid(const char *signature, const char **names, int *flags) {
+        enum {
+                first           = 1<<0,
+                have_names      = 1<<1,
+                alone           = 1<<2,
+        };
```

Comment:
the whole function should be shortcut if no names are specified... no need to parse the signature string then

Reply:
these flags are not obvious, they surely could use some brief comments explaining what they indicate